packages/cms/pages/resources/edit/edit.tsx

```diff
@@ -0,0 +1,353 @@
+      </Title>
+      <TitleUnderline />
+      <EuiSpacer size="m" />
+      <Contents close={close}>{children}</Contents>
+    </Item>
+  )
+}
+
+const UpdateResourceSidebar: React.FunctionComponent<{
+  resourceData: any
+  resource: ResourceContract | undefined
+}> = ({ resource, resourceData }) => {
+  const [close, setClose] = useState(false)
+  const onCloseSideBar = () => setClose(!close)
+
+  return (
+    <Sidebar close={close}>
+      <SidebarCollapseExpandIcon close={close} onClick={onCloseSideBar}>
+        {close ? (
+          <EuiIcon size="s" type="arrowLeft" onClick={onCloseSideBar} />
+        ) : (
+          <EuiIcon size="s" type="arrowRight" onClick={onCloseSideBar} />
+        )}
+      </SidebarCollapseExpandIcon>
+
+      <SidebarItem
+        title={`${resource?.name?.toUpperCase()} INFORMATION`}
+        sidebarClose={close}
+      >
+        <Content>
+          <EuiText size="s">Created by</EuiText>
+          <ValueText size="s">-</ValueText>
+        </Content>
+        <Content>
+          <EuiText size="s">Last updated</EuiText>
+          <ValueText size="s">
+            {moment(resourceData?.updatedAt).fromNow()}
+          </ValueText>
+        </Content>
+      </SidebarItem>
+
+      <EuiSpacer size="l" />
+
+      <SidebarItem title="LOCALE" sidebarClose={close}>
+        <Content>
+          <ValueText size="s">en-US</ValueText>
+        </Content>
+      </SidebarItem>
+    </Sidebar>
+  )
+}
+
+const PublishAndSaveToDraftContainer = styled.div`
+  gap: 0.75rem;
+  display: flex;
+  align-items: center;
+`
+const TitleAndBackButtonContainer = styled.div`
+  gap: 0.75rem;
+  display: flex;
+  align-items: center;
+`
+const PageWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  overflow-y: auto;
+  padding: 40px;
+`
+const PageContent = styled.div`
+  // width: 50%;
+  width: 650px;
+  margin: 0px auto;
+`
+
+export const EditResource: React.FunctionComponent = () => {
+  // const theme = useEuiTheme()
+  const { push, goBack } = useHistory()
+  const [activeField, setActiveField] = useState<FieldContract>()
+  const {
+    findResource,
+    fetchResourceData,
+    updateResource,
+    resource
+  } = useResourceStore()
+  const { resource: resourceSlug, id: resourceId = '' } = useParams<{
+    resource: string
+    id: string
+  }>()
+  const { toast } = useToastStore()
+  const [booted, setBooted] = useState(false)
+
+  const {
+    form,
+    setForm,
+    errors,
+    submit,
+    loading,
+    setValue
+  } = useForm<AbstractData>({
+    defaultValues: {},
+    onSubmit: (form: AbstractData) =>
+      updateResource(parseInt(resourceId), form),
+    onSuccess: () => {
+      toast(
+        'Updated',
+        <p>{resource?.name.toLowerCase()} have been updated successfully</p>
+      )
+
+      push(window.Tensei.getPath(`resources/${resource?.slugPlural}`))
+    }
+  })
+
+  const getData = async () => {
+    const [data, error] = await fetchResourceData(parseInt(resourceId))
```

Comment:
We can't `parseInt` because we also use Mongodb, where the primary key is not an integer. Also, some projects will use UUIDs as primary keys, so we can't parse it as an integer. has to be a string instead.